### ISSUE TITLE
Effects: Instantiating Effects multiple times, only first instance is run

### ISSUE THREAD

**Issue by dummdidumm** [ngrx/platform#1246]
## Minimal reproduction of the bug/regression with instructions:

When instantiating an effect multiple times and adding it to EffectSources, only the first instance is run, all others are ignored.

https://stackblitz.com/edit/angular-cfqrvf

## Expected behavior:

Adding instances through EffectSources.addEffects should work when providing multiple instances of the same class, so this should work:

```
class ConfigurableEffect {
  @Effect({ dispatch: false })
  doSth$ = timer(1000).pipe(
    tap(() => console.log(this.message))
  );

  constructor(private message: string) { }
}

@NgModule({
  imports: [BrowserModule, FormsModule, StoreModule.forRoot(() => true), EffectsModule.forRoot([])],
  declarations: [AppComponent, HelloComponent],
  bootstrap: [AppComponent]
})
export class AppModule {
  constructor(effectSources: EffectSources) {
    effectSources.addEffects(new ConfigurableEffect("configA"));
    effectSources.addEffects(new ConfigurableEffect("configB"));
    // -> should run both "configA" and "configB" after 1 second, but only "configA" is run
  }
}
```

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):

Latest

## Other information:

This is essentially blocking us from using effects in a configurable way.

## I would be willing to submit a PR to fix this issue

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by timdeschryver** [comment]
If I'm correct I think this is intended. 

For ref:
[modules/effects/src/effect_sources.ts#L32](https://github.com/ngrx/platform/blob/master/modules/effects/src/effect_sources.ts#L32
)

[effect_sources.spec.ts#L94](https://github.com/ngrx/platform/blob/master/modules/effects/spec/effect_sources.spec.ts#L94)

**Comment by dummdidumm** [comment]
I think it's only intended because EffectsModule.forFeature is a token with "multi: true", so when adding feature effects, it somehow has to be determined which effects are already running so you don't run them again https://github.com/ngrx/platform/blob/master/modules/effects/src/effects_feature_module.ts. The determination "oh this is a duplicate" is done by looking at the prototype, which I think is wrong. There must be a better way so that adding multiple instances like shown above works.

**Comment by dummdidumm** [comment]
I did some digging and found that `toActions` inside `EffectsSource` has to be rewritten to allow it. I made a PR to support the case highlighted in here.

**Comment by timdeschryver** [comment]
@binu101 you can install it via the  nightly builds.
https://github.com/ngrx/platform/tree/master/docs/effects#nightly-builds

**Comment by binu101** [comment]
Just installed 6.1.1 (in which this should be fixed) but i still have the getSourceByInstance groupBy pipe in my @ngrx/effects source code..

**Comment by timdeschryver** [comment]
It isn't available in 6.1.1
It will be in 7.x
